x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteExpiredDataAction.java

```diff
@@ -164,7 +172,25 @@ void deleteExpiredData(Iterator<MlDataRemover> mlDataRemoversIterator,
             if (haveAllPreviousDeletionsCompleted) {
                 logger.info("Completed deletion of expired ML data");
             } else {
-                logger.info("Halted deletion of expired ML data until next invocation");
+                if (isTimedOutSupplier.get()) {
+                    String msg = "Deleting expired ML data was cancelled after the timeout period of [" +
+                        MlDataRemover.DEFAULT_MAX_DURATION  + "] was exceeded. " +
+                        "The setting [xpack.ml.nightly_maintenance_requests_per_second] " +
+                        "controls the deletion rate, consider increasing the value to assist in pruning old data";
+                    logger.warn(msg);
+
+                    if (Strings.isNullOrEmpty(request.getJobId())
+                        || Strings.isAllOrWildcard(new String[]{request.getJobId()})
```

Comment:
There is a `Strings#isAllOrWildcard(String)`
```suggestion
                        || Strings.isAllOrWildcard(request.getJobId())
```